File: frontend/app/api/services/mail.ts
import http from "./httpInstance";

export type MailSendRequest = {
	subject: string;
	messageBody: string;
	priority?: string;
	recipients: string[];
	cc?: string[];
	bcc?: string[];
	attachments?: string[];
};

class MailService {
	/** Send mail via the backend SMTP endpoint (credentials stay on the API host). */
	async send(
		subject: string,
		messageBody: string,
		priority: string,
		recipients: string[],
		cc: string[],
		bcc: string[],
		attachments: string[]
	) {
		const body: MailSendRequest = {
			subject,
			messageBody,
			priority,
			recipients,
			cc,
			bcc,
			attachments,
		};

		await http.post("/mail", body);
	}
}

const mailService = new MailService();
export default mailService;

File: backend/app/schemas/mail.py
from pydantic import BaseModel, Field


class MailSendRequest(BaseModel):
    subject: str = Field(min_length=1, max_length=500)
    message_body: str = Field(alias="messageBody", min_length=1)
    priority: str = "Normal"
    recipients: list[str] = Field(min_length=1, max_length=5)
    cc: list[str] = Field(default_factory=list)
    bcc: list[str] = Field(default_factory=list)
    attachments: list[str] = Field(default_factory=list)

    model_config = {"populate_by_name": True}
